### PR TITLE
Use local variables, don't modify a: dict

### LINKED ISSUE #197: Vim 8.1 (1-899): Illegal Variable Name a:attr, a:guisp
Hi and thanks for this great plugin in addition to the terminal colors:

I have not compiled vim from source in a while (so this could be my fault if I have overlooked a missing feature), but I did last night on my windows 10 machine. Everything works fine after the compilation except this plugin throws errors when opening vim. The messages are as follows:

image

image

I am using the base16-eighties colorscheme and this is my vimrc below:

**vimrc**
```
call dein#add('chriskempson/base16-vim')

if filereadable(expand("~/.vimrc_background"))
  "per the docs only enable this if the ~/.cofig/base16/colortest does not show blue for 17-21
  let base16colorspace=256
  source ~/.vimrc_background
endif
```

**vimrc_background**
```
if !exists('g:colors_name') || g:colors_name != 'base16-eighties'
  colorscheme base16-eighties
endif
```
When i remove this from my vimrc vim opens and works fine

### PR DESCRIPTION
**Pull request by danielwe**
Fixes #197

### CODE REVIEW

**Comment by gylaz** [comment]
Is anyone still maintaining this repo?

**Comment by ptgamr** [comment]
Please merge ...

**Comment by wrboyce** [comment]
Ping @chriskempson can you merge this please?

**Comment by howardburgess** [comment]
Thanks for working on this @danielwe. I saw a bunch of color changes in the diffs too (eg https://github.com/chriskempson/base16-vim/pull/198/files#diff-00c9ae40dd5792474ac4d9ebc5efe90eR25). Did these creep in by mistake?

**Comment by danielwe** [comment]
I saw those too. My only deliberate change was to the template code. The way the base16 build system works, it pulls the latest theme definitions from their respective repos (via https://github.com/chriskempson/base16-templates-source) and builds the package by applying the template to each of them. So what's happened is that some theme authors have updated the colors used in their themes since the previous commit to base16-vim.

**Comment by howardburgess** [comment]
Ah ok, thought it would be something like that 👍

**Comment by mclang** [comment]
Hello

I just updated new vim last week and started getting these errors... I really hope @chriskempson finds the time to merge this soon!

**Comment by ptrxyz** [comment]
Any plans to merge this?

File: colors/base16-3024.vim [hunk 1]
```diff
@@ -150,8 +150,8 @@ let g:colors_name = "base16-3024"
 " Highlighting function
 " Optional variables are attributes and guisp
 function! g:Base16hi(group, guifg, guibg, ctermfg, ctermbg, ...)
-	let a:attr = get(a:, 1, "")
-	let a:guisp = get(a:, 2, "")
+	let l:attr = get(a:, 1, "")
+	let l:guisp = get(a:, 2, "")
```
Comment (blueyed) [inline colors/base16-3024.vim]: Maybe you can include https://github.com/chriskempson/base16-vim/pull/203 here, i.e. change this to use spaces.

**Comment by iovis9** [comment]
Any news?

File: colors/base16-3024.vim (hunk 1 above)
Comment (danielwe) [reply colors/base16-3024.vim]: Good catch!

**Comment by blueyed** [comment]
@iovis9 
No - have you heard something? ;)
(AFAICS this appears to be unmaintained unfortunately)

**Comment by dylan-chong** [comment]
I am getting this issue on normal vim

Screenshot 2019-06-03 at 11 43 56 AM

But it works fine on neovim

Screenshot 2019-06-03 at 11 44 37 AM

And also mac vim

Screenshot 2019-06-03 at 11 46 14 AM

I am using the materia theme

**Pull request merged**